tests/unit/test_signers.py

```diff
@@ -1168,3 +1175,37 @@ def test_custom_region(self):
         self.assertIn(region, result)
         # The hostname won't be changed even if a different region is specified
         self.assertIn(hostname, result)
+
+
+@pytest.mark.parametrize(
+    'request_method, content_type_present',
+    [
+        ('GET', False),
+        ('HEAD', False),
+        ('OPTIONS', False),
+        ('POST', True),
+        ('PUT', True),
+        ('DELETE', True),
+        (None, True),
+    ],
+)
+@mock.patch('botocore.signers.RequestSigner.generate_presigned_url')
+def test_generate_presigned_url_content_type_removal(
+    generate_presigned_url_mock,
+    polly_client,
+    request_method,
+    content_type_present,
+):
+    polly_client.generate_presigned_url(
+        'synthesize_speech',
+        Params={
+            'OutputFormat': 'mp3',
+            'Text': 'Hello world!',
+            'VoiceId': 'Joanna',
+        },
+        HttpMethod=request_method,
+    )
+    assert (
+        'Content-Type'
+        in generate_presigned_url_mock.call_args[1]['request_dict']['headers']
+    ) == content_type_present
```

Comment:
Do we need a mock for this? It should just be in the final URL and that's the part we're actually concerned about.

Reply:
Yeah tests were failing earlier since the signer needs to use credentials. I initially was checking the URL, but I chose to mimic this test suite which uses a mock. I'll see if I can just provide the client with dummy credentials instead.

Reply:
Yeah tests were failing earlier since the signer needs to use credentials. I initially was checking the URL, but I chose to mimic this test suite which uses a mock. I'll see if I can just provide the client with dummy credentials instead.

Reply:
I think we already have [credentials set on the test class](https://github.com/boto/botocore/pull/2944/files#diff-c93558930e421bfe0fd7767dad14f92702b68bf878b550b3d231ccc09e7d47e8R51) in this test suite. We can do the same thing for your client fixture.